File: php/src/Framework/FrameworkRequest.php
<?php
declare(strict_types=1);

namespace Demo\App\Framework;

final readonly class FrameworkRequest
{
    public const string METHOD_GET = 'GET';
    public const string METHOD_POST = 'POST';
    public const string METHOD_PUT = 'PUT';
    public const string METHOD_PATCH = 'PATCH';

    public function __construct(private string $method, private string $path, private array $content)
    {
    }

    public function method(): string
    {
        return $this->method;
    }

    public function path(): string
    {
        return $this->path;
    }

    public function pathStart(): string
    {
        return dirname($this->path());
    }

    public function getIdPath(): string
    {
        return basename(parse_url($this->path(), PHP_URL_PATH));
    }

    public function content(): array
    {
        return $this->content;
    }
}

File: kotlin/src/main/kotlin/framework/FrameworkRequest.kt
package framework

data class FrameworkRequest(val method: String, val path: String, val content: Map<String, String>)
{
    companion object {
        const val METHOD_GET = "GET"
        const val METHOD_POST = "POST"
        const val METHOD_PUT = "PUT"
        const val METHOD_PATCH = "PATCH"
    }

    fun pathStart(): String {
        return path.substringBefore("/")
    }

    fun getIdPath(): String {
        return path.substringAfter("/")
    }
}
